pkg/ibmvpc/sg_analysis.go

```diff
@@ -291,18 +292,43 @@ func (sga *SGAnalyzer) prepareAnalyzer(sgMap map[string]*SecurityGroup, currentS
 }
 
 func (sga *SGAnalyzer) AllowedConnectivity(target string, isIngress bool) *common.ConnectionSet {
-	ipb := common.NewIPBlockFromCidrOrAddress(target)
-
-	var analyzedConns *ConnectivityResult
-	if isIngress {
-		analyzedConns = sga.ingressConnectivity
-	} else {
-		analyzedConns = sga.egressConnectivity
-	}
+	analyzedConns, ipb := sga.getAnalyzedConnsIPB(target, isIngress)
 	for definedTarget, conn := range analyzedConns.allowedconns {
 		if ipb.ContainedIn(definedTarget) {
 			return conn
 		}
 	}
 	return vpcmodel.NoConns()
 }
+
+// rulesInConnectivity list of SG rules contributing to the connectivity
+func (sga *SGAnalyzer) rulesInConnectivity(target string, isIngress bool) []int {
+	analyzedConns, ipb := sga.getAnalyzedConnsIPB(target, isIngress)
+	for definedTarget, rules := range analyzedConns.contribRules {
+		if ipb.ContainedIn(definedTarget) {
+			return rules
+		}
+	}
+	return nil
+}
+
+func (sga *SGAnalyzer) getAnalyzedConnsIPB(target string, isIngress bool) (res *ConnectivityResult, ipb *common.IPBlock) {
+	ipb = common.NewIPBlockFromCidrOrAddress(target)
+	if isIngress {
+		return sga.ingressConnectivity, ipb
+	}
+	return sga.egressConnectivity, ipb
+}
+
+// StringRules returns a string with the details of the specified rules
+func (sga *SGAnalyzer) StringRules(rules []int) string {
+	var strRules string
+	for _, ruleIndex := range rules {
+		strRule, _, _, err := sga.getSGRule(ruleIndex)
+		if err != nil {
+			return ""
+		}
```

Comment:
Since its a printing function, I preferred to have it return a clean string. It is impossible (or nearly impossible) that we will get here if the relevant `getSGRule`  fails on one of the relevant rules. But I agree this is not a clear cut.